crates/core/src/db/relational_db.rs

```diff
@@ -1403,6 +1403,20 @@ impl RelationalDB {
         Ok(rows_deleted)
     }
 
+    /// Clear all rows from all view tables without dropping them.
+    pub fn clear_all_views(&self, tx: &mut MutTx) -> Result<(), DBError> {
+        for table_id in tx
+            .iter(ST_VIEW_ID)?
+            .map(StViewRow::try_from)
+            .collect::<Result<Vec<_>, _>>()?
+            .into_iter()
+            .filter_map(|row| row.table_id)
+        {
+            tx.clear_table(table_id)?;
+        }
+        Ok(())
```

Comment:
Let's move the actual logic into the datastore and have this method forward to that.